From 7b8855233c0c969463384d5ec8b3a8b4e7f70a31 Mon Sep 17 00:00:00 2001
From: Karsten Blees <blees@dcon.de>
Date: Thu, 5 Jul 2012 10:27:11 +0200
Subject: [PATCH 15/17] msys.dll: support ALT+NUMPAD console input

The msys console handler currently ignores all key up events.

Support ALT+NUMPAD console input by accepting characters transfered in key
up events of the ALT (VK_MENU) key.

Signed-off-by: Karsten Blees <blees@dcon.de>
---
 msys/rt/src/winsup/cygwin/fhandler_console.cc |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/msys/rt/src/winsup/cygwin/fhandler_console.cc b/msys/rt/src/winsup/cygwin/fhandler_console.cc
index 724db03..1c3de60 100644
--- a/msys/rt/src/winsup/cygwin/fhandler_console.cc
+++ b/msys/rt/src/winsup/cygwin/fhandler_console.cc
@@ -298,10 +298,11 @@ fhandler_console::read (void *pv, size_t buflen)
 	      break;
 	    }
 
-	  if (!input_rec.Event.KeyEvent.bKeyDown)
+#define wch (input_rec.Event.KeyEvent.uChar.UnicodeChar)
+
+	  if (!input_rec.Event.KeyEvent.bKeyDown && (wch == 0 || virtual_key_code != VK_MENU))
 	    continue;
 
-#define wch (input_rec.Event.KeyEvent.uChar.UnicodeChar)
 #define ALT_PRESSED (LEFT_ALT_PRESSED | RIGHT_ALT_PRESSED)
 #define CTRL_PRESSED (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)
 
-- 
1.7.10.rc1.5708.g87c2b

